lib/azure/armrest/resource_group_based_service.rb

```diff
@@ -233,15 +235,31 @@ def validate_resource(name)
       #
       def build_url(resource_group = nil, *args)
         url = File.join(configuration.environment.resource_url, build_id_string(resource_group, *args))
-        url << "?api-version=#{@api_version}"
       end
 
       def build_id_string(resource_group = nil, *args)
         id_string = File.join('', 'subscriptions', configuration.subscription_id)
         id_string = File.join(id_string, 'resourceGroups', resource_group) if resource_group
         id_string = File.join(id_string, 'providers', @provider, @service_name)
-        id_string = File.join(id_string, *args) unless args.empty?
-        id_string
+
+        query = "?api-version=#{@api_version}"
+
+        args.each do |arg|
+          if arg.kind_of?(Hash)
+            arg.each do |key, value|
+              if key.to_s.casecmp('top').zero?
+                query << "&$top=#{value}"
+              else
+                query << "&$filter=" unless query.include?("$filter")
```

Comment:
Not if they submit multiple filters. Though, come to think of it, that wouldn't actually work at the moment because we need an "and".